### PR TITLE
Review the Docker annotation BAL file

### PR DESCRIPTION
**Pull request by praneesha**
## Purpose
This is review and fix typos and issues of [1].

[1] https://github.com/ballerina-platform/ballerina-lang/blob/master/stdlib/docker/src/main/ballerina/src/docker/annotation.bal

Fixes https://github.com/ballerina-platform/ballerina-lang/issues/19684

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: stdlib/docker/src/main/ballerina/src/docker/annotation.bal [hunk 1]
```diff
@@ -16,22 +16,22 @@
...
+# + cmd - CMD value for the generated Docker file. Default is `CMD java -jar ${APP} [--b7a.config.file=${CONFIG_FILE}] [--debug]`.
+# + enableDebug - Enable Ballerina debugging. Default is `false`.
+# + debugPort - Ballerina remote debugging port. Default is `5005`.
 # + dockerAPIVersion - Docker API version.
-# + dockerHost - Docker host IP and docker PORT. (e.g minikube IP and docker PORT).
-# Default is to use DOCKER_HOST environment variable.
-# If DOCKER_HOST is unavailable, use `"unix:///var/run/docker.sock"` for Unix or use `"npipe:////./pipe/docker_engine"` for Windows 10 or use `"localhost:2375"`.
-# + dockerCertPath - Docker certificate path. Default is to use `"DOCKER_CERT_PATH"` environment variable.
+# + dockerHost - Docker host IP and Docker port (e.g,. Minikube IP and Docker port).
+# Default is to use the DOCKER_HOST environment variable.
```
Comment (anupama-pathirage) [inline stdlib/docker/src/main/ballerina/src/docker/annotation.bal]:
```suggestion
# Default is to use the `DOCKER_HOST` environment variable.
```
```suggestion
# Default is to use the DOCKER_HOST environment variable.
```

File: stdlib/docker/src/main/ballerina/src/docker/annotation.bal [hunk 2]
```diff
@@ -16,22 +16,22 @@
...
+# If DOCKER_HOST is unavailable, use `"unix:///var/run/docker.sock"` for Unix, use `"npipe:////./pipe/docker_engine"` for Windows 10, or use `"localhost:2375"`.
```
Comment (anupama-pathirage) [inline stdlib/docker/src/main/ballerina/src/docker/annotation.bal]:
```suggestion
# If `DOCKER_HOST` is unavailable, use `"unix:///var/run/docker.sock"` for Unix, use `"npipe:////./pipe/docker_engine"` for Windows 10, or use `"localhost:2375"`.
```

File: stdlib/docker/src/main/ballerina/src/docker/annotation.bal (hunk 1 above)
Comment (anupama-pathirage) [inline stdlib/docker/src/main/ballerina/src/docker/annotation.bal]:
```suggestion
# Default is to use the `DOCKER_HOST` environment variable.
```

File: stdlib/docker/src/main/ballerina/src/docker/annotation.bal [hunk 3]
```diff
@@ -16,22 +16,22 @@
...
-# + buildImage - Enable building docker image. Default value is `true`.
-# + push - Enable pushing docker image to registry. Field `buildImage` must be set to `true` to be effective. Default value is `false`.
-# + cmd - Value for CMD for the generated Dockerfile. Default is `CMD java -jar ${APP} [--b7a.config.file=${CONFIG_FILE}] [--debug]`.
-# + enableDebug - Enable ballerina debug. Default is `false`.
-# + debugPort - Ballerina remote debug port. Default is `5005`.
+# + username - Username for the Docker registry.
+# + password - Password for the Docker registry.
+# + baseImage - Base image to create the Docker image. Default value is `"openjdk:8-jre-alpine"`.
+# + buildImage - Enable building the Docker image. Default value is `true`.
+# + push - Enable pushing the Docker image to registry. The value of the `buildImage` field must be set to `true` to be effective. Default value is `false`.
```
Comment (anupama-pathirage) [inline stdlib/docker/src/main/ballerina/src/docker/annotation.bal]:
```suggestion
# + push - Enable pushing the Docker image to the registry. The value of the `buildImage` field must be set to `true` to be effective. Default value is `false`.
```

File: stdlib/docker/src/main/ballerina/src/docker/annotation.bal (hunk 1 above)
Comment (pubudu91) [reply stdlib/docker/src/main/ballerina/src/docker/annotation.bal]: Default behavior is to use the `DOCKER_HOST` environment variable

File: stdlib/docker/src/main/ballerina/src/docker/annotation.bal [hunk 4]
```diff
@@ -16,22 +16,22 @@
...
+# + debugPort - Ballerina remote debugging port. Default is `5005`.
 # + dockerAPIVersion - Docker API version.
-# + dockerHost - Docker host IP and docker PORT. (e.g minikube IP and docker PORT).
-# Default is to use DOCKER_HOST environment variable.
-# If DOCKER_HOST is unavailable, use `"unix:///var/run/docker.sock"` for Unix or use `"npipe:////./pipe/docker_engine"` for Windows 10 or use `"localhost:2375"`.
-# + dockerCertPath - Docker certificate path. Default is to use `"DOCKER_CERT_PATH"` environment variable.
+# + dockerHost - Docker host IP and Docker port (e.g,. Minikube IP and Docker port).
+# Default is to use the DOCKER_HOST environment variable.
+# If DOCKER_HOST is unavailable, use `"unix:///var/run/docker.sock"` for Unix, use `"npipe:////./pipe/docker_engine"` for Windows 10, or use `"localhost:2375"`.
+# + dockerCertPath - Docker certificate path. Default is to use the `"DOCKER_CERT_PATH"` environment variable.
```
Comment (pubudu91) [inline stdlib/docker/src/main/ballerina/src/docker/annotation.bal]:
```suggestion
# + dockerCertPath - Docker certificate path. Default behavior is to use the `DOCKER_CERT_PATH` environment variable.
```

File: stdlib/docker/src/main/ballerina/src/docker/annotation.bal (hunk 1 above)
Comment (praneesha) [reply stdlib/docker/src/main/ballerina/src/docker/annotation.bal]: > Default behavior is to use the `DOCKER_HOST` environment variable

Added.

**Comment by pubudu91** [comment]
Note that this same fix should be sent to `ballerina-1.1.x` branch

**Comment by praneesha** [comment]
> Note that this same fix should be sent to `ballerina-1.1.x` branch

This is adressed via https://github.com/ballerina-platform/ballerina-lang/pull/20597.

**Pull request merged**